### ISSUE TITLE
Not a known attribute of module error under shadowing import

### ISSUE THREAD

**Issue by MathiasSven** [microsoft/pyright#9267]
**Describe the bug**
Perhaps not the right title, but I just noticed this issue while working with a 3rd party library. I made an MRE bellow of what the library essentially had. 

**To Reproduce**
Project Structure:
```
.
├── foo
│   ├── __init__.py
│   ├── sub1.py
│   └── sub2.py
└── main.py
```
```python
# foo.__init__.py
from . import sub1
from .sub2 import *
```
```python
# foo.sub1.py
a = 1
b = 2
```
```python
# foo.sub2.py
from . import sub1    # Removing this line makes the error go away
from .sub1 import a
```
```python
# main.py
import foo
import foo.sub1
import foo.sub1 as _sub1

print(
    foo.sub1.a,  # No errors
    foo.sub1.b,  # "b" is not a known attribute of module "foo"
    _sub1.b      # No errors
)
```

**VS Code extension or command-line**
Error appears on both the extension via VSCode and when calling the command line tool.

**Comment by erictraut** [comment]
That's a tricky case to handle statically, but I agree that it's a bug. Thanks for taking the time to put together clear repro steps.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
